### DISCUSSION TITLE
How to make referenced assembly (dll) visible in other assemblies

### DISCUSSION THREAD

**Discussion by kefyru** [dotnet/msbuild#8126] [Q&A]
I have library `MyLib`:
```xml
<Project Sdk="Microsoft.NET.Sdk">
    <PropertyGroup>
        <TargetFramework>net6.0</TargetFramework>
        <RootNamespace>MyLib</RootNamespace>
    </PropertyGroup>
    <ItemGroup>
        <Reference Include="SomeUtil">
            <HintPath>..\utils\SomeUtil.dll</HintPath>
        </Reference>
    </ItemGroup>
</Project>
```
`SomeUtil.dll` is .net assembly built with external tool.
In `MyLib` i can access all members of `SomeUtil.dll`
When i reference `MyLib` to another assembly, i have no access to `SomeUtil.dll`.
I can reference it manually, but is there any way to reference it automatically?

**Comment by JanKrivanek** [comment]
Hi @kefyru - the behavior you are observing is by design and actually desired. The `SomeUtil.dll` should be explicitly referenced by all the projects that rely on it.

Access to transitive dependencies can reinforce bad design practices by allowing breaking of layering of the intended architecture. All projects should declare their dependencies explicitly.

The SDK projects opened access to transitive _project_ references - but this was provided mainly for compiler and analyzers purposes. End user projects are not recommended to rely on and (ab)use the behavior (in fact - one can, and probably should - set `ReferenceOutputAssembly` metadata to `false` in order to prevent breaking through architecture layers).

More details on controling dependencies behavior can be found in our docs: https://github.com/dotnet/msbuild/blob/main/documentation/wiki/Controlling-Dependencies-Behavior.md
